core/src/main/java/io/camunda/migrator/Pagination.java

```diff
@@ -88,25 +96,58 @@ public List<T> toList() {
     return list;
   }
 
+  public Map<String, Map<String, Object>> toVariableMap() {
```

Comment:
🙃  This could be simplified like this: 
```
public Map<String, Map<String, Object>> toVariableMap() {
    Map<String, Map<String, Object>> result = new HashMap<>();
    processVariables((variable, typedValue) -> {
      result
          .computeIfAbsent(variable.getActivityInstanceId(), k -> new HashMap<>())
          .putAll(getVariableResult(variable, typedValue));
    });
    return result;
```